### ISSUE TITLE
[turborepo] "error in yarn.lock" of Yarn Canary since v1.9.4

### ISSUE THREAD

**Issue by robaca** [vercel/turbo#5220]
### What version of Turborepo are you using?

1.10.1

### What package manager are you using / does the bug impact?

Yarn v2/v3 (node_modules linker only)

### What operating system are you using?

Linux

### Describe the Bug

We are currently using Turborepo in a monorepo in combination with Yarn Canary (v4-rc45).
Starting with version 1.9.4 of Turborepo, it complains about an error in yarn.lock:

```
 WARNING  Issues occurred when constructing package graph. Turbo will function, but some features may not be available:  1 error occurred:
	* error in yarn.lock: Unable to decode yarn.lock: ParseError: Unknown err
```

I already checked line lengths (longest is 630) and ran `yamllint` against the lockfile, but without any error.

Downgrading to yarn 3.6.0 or Turborepo 1.9.3, I don't get the error any more.

### Expected Behavior

No warning.

### To Reproduce

Our project is not publicly available. I could send the lockfile via PM.

### Reproduction Repo

a private repo

**Comment by belgattitude** [comment]
You can find a reproduction in this repository: https://github.com/belgattitude/nextjs-monorepo-example

It's based on yarn 4.0.0-rc.45 and uses latest turbo 1.10.3. 

To install an example:

```bash
git clone https://github.com/belgattitude/nextjs-monorepo-example.git
cd nextjs-monorepo-example
corepack enable
yarn install
yarn turbo lint
```

You'll see

```
WARNING  Issues occurred when constructing package graph. Turbo will function, but some features may not be available:  1 error occurred:
	* error in yarn.lock: Unable to decode yarn.lock: ParseError: Unknown err

• Packages in scope: @your-org/api-gateway, @your-org/common-i18n, @your-org/core-lib, @your-org/db-main-prisma, @your-org/eslint-config-bases, @your-org/ts-utils, @your-org/ui-lib, nextjs-app, remix-app, vite-app
• Running lint in 10 packages
• Remote caching disabled
@your-org/ui-lib:lint: cache miss, executing 7bb7d913ca605998
...
```

The field `__metadata.version` allows to detect which version is in use. 

```yaml
__metadata:
  version: 7  for yarn 4.0.0-rc+, 6  for yarn 3+
```

I've tried a downgrade to 3.6.0 to see what are the real differences between the lockfile V7 et V6...

The PR is here: https://github.com/belgattitude/nextjs-monorepo-example/pull/3942

The only thing that seems relevant for turbo is the prefix in front of the version (I guess to disambiguate source of install: npm, local patch, git...), but I assume they might have something else than just `npm:`

```diff
dependencies:
+    buffer-crc32: "npm:~0.2.3"
+   fd-slicer: "npm:~1.1.0"
-    buffer-crc32: ~0.2.3
-    fd-slicer: ~1.1.0
```

Couldn't find documentation on yarn website (yet)

**Comment by mignotju** [comment]
Hello! I had the same issue, upgrading **turbo** to `1.10.6` solved it ✅

**Comment by belgattitude** [comment]
@mignotju I can confirm as well. 

I believe it was fixed in 1.10.4 by @chris-olszewski in that PR https://github.com/vercel/turbo/pull/5255. 

Tested with yarn 4.0.0-rc.45 and 4.0.0-rc.46 for the changes detection but not for docker (yet)

I feel it can be closed

**Comment by mehulkar** [comment]
Thanks for confirming, closing this down!
